src/main/java/org/firmata4j/firmata/FirmataDevice.java

```diff
@@ -74,25 +61,31 @@
     private final Map<Byte, FirmataI2CDevice> i2cDevices = new HashMap<>();
     private volatile Map<String, Object> firmwareInfo;
     private volatile Map<Integer, Integer> analogMapping;
-    private static final long TIMEOUT = 15000L;
-    private static final int DELAY = 15;
-    private static final Logger LOGGER = LoggerFactory.getLogger(FirmataDevice.class);
 
     /**
      * Constructs FirmataDevice instance on specified port.
      *
      * @param portName the port name the device is connected to
      */
     public FirmataDevice(String portName) {
-        this.port = new SerialPort(portName);
+        this(new SerialFirmataTransport(portName));
+    }
+
+    /**
+     * Constructs FirmataDevice instance using the specified device;
+     *
+     * @param transport the hardware device to use
+     */
+    public FirmataDevice(AbstractFirmataTransport transport) {
+        this.transport = transport;
+        EventPublisher.setListener(this);
```

Comment:
`EventPublisher` implemented the way that we cannot have more than one instance of `FirmataDevice` in an application. If the application's intention is operate, let's say, two boards, it is impossible to do because the second instance rewrites the listener.

I think that `EventPublisher` makes redundant point of inderection. The only thing it does is redirecting events from `FrmataParser` to `FirmataDevice`. It is possible to do dirrectly.

I suggest the following way of thinking:
- `FirmataDevice` is an integral part that knows about `FirmataParser` and `FirmataTransportInterface` instances.  `FirmataDevice` gets provided with those instances and if not, instantiates some default implementations.
- `FirmataDevice` registers `FirmataParser` instance in `FirmataTransportInterface` instance.
- `FirmataDevice` registers itself in `FirmataParser` instance.